tests/native_tests/native_macros.hy

```diff
@@ -329,3 +330,137 @@
     (except [e SystemExit]
       (assert (= (str e) "42"))))
   (setv --name-- oldname))
+
+#@(pytest.mark.xfail
+    (defn test-macro-module-expansions []
```

Comment:
I think this test is wrong: `(nonlocal-test-macro 1)` should have been expanded before `(setv module-name-var "tests.resources.macros")` was evaluated, because the former is a macro expansion while the latter is ordinary code and they're in the same file. See my comments on #1414.

Reply:
Yeah, that's why it's currently an expected fail, but I'm open to making it work &mdash; e.g. with the incremental compile-and-eval approach.  If there aren't any objections to that design change, I can put that change higher up on my priorities.

By the way, an incremental approach would drastically simplify debugging!

Reply:
Such an incremental approach would not allow one to expand all macros before compiling to bytecode or translating to Python.

Reply:
Why's this marked resolved? The test is still there.